maggy/maggy/api/routes_pr_review.py

```
@@ -0,0 +1,109 @@
+    dry = cfg.review.default_dry_run if body.dry_run is None else body.dry_run
+    repo_path = body.repo_path or cfg.review.repo_paths.get(f"{body.owner}/{body.repo}")
+    logs: list[str] = []
+    try:
+        out = await run_review(
+            body.owner, body.repo, body.pr_number,
+            dry_run=dry, repo_path=repo_path, on_log=logs.append, token=token,
+        )
+    except SystemExit as exc:        # pipeline raises SystemExit for config problems
+        raise HTTPException(400, str(exc))
+    except Exception as exc:          # noqa: BLE001 — surface the failure, don't 500 opaquely
+        raise HTTPException(502, f"Review failed: {type(exc).__name__}: {exc}")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)